### PR TITLE
[CL][Internal Review] CLI and query for incentive records per pool

### PR DESCRIPTION
**Pull request by stackman27**
Closes: [#5085](https://github.com/osmosis-labs/osmosis/issues/5085)

## What is the purpose of the change
- add query for incentives_records

## Testing and Verifying
local osmosis tested 

```
MacBook-Pro-26:osmosis sishirg27$ osmosisd q concentratedliquidity incentive-records 1
incentive_records:
- incentive_creator_addr: osmo12smx2wdlyttvyzvzg54y2vnqwq2qjateuf7thj
  incentive_denom: uion
  incentive_record_body:
    emission_rate: "100.000000000000000000"
    remaining_amount: "100000000.000000000000000000"
    start_time: "2023-05-16T21:20:30.203746449Z"
  min_uptime: 0.000000001s
  pool_id: "1"
pagination: null
```

## Documentation and Release Note
n/a

### CODE REVIEW

**Review by p0mvn — APPROVED** [review]
LGTM

**Review by mattverse — APPROVED** [review]
core logic lgtm, does osmocli automatically handle pagination by default?

Would like to merge this after seeing proof or log of running this query with pagination

**Review by mattverse — APPROVED** [review]

**Comment by mattverse** [comment]
Sorry about the messy reviews here, didn't know github does not allow changing reviews once posted -ㅡ-

**Review by czarcas7ic — APPROVED** [review]
This LGTM, but I imagine it will need to be paginated

File: x/concentrated-liquidity/incentives.go [hunk 1]
```diff
@@ -530,6 +535,64 @@ func (k Keeper) GetAllIncentiveRecordsForPool(ctx sdk.Context, poolId uint64) ([
...
+
+}
+
+// GetPaginatedIncentiveRecordsForPool
+func (k Keeper) GetPaginatedIncentiveRecordsForPool(ctx sdk.Context, poolId uint64, pagination *query.PageRequest) ([]*codectypes.Any, *query.PageResponse, error) {
+	store := ctx.KVStore(k.storeKey)
+	incentiveRecordStore := prefix.NewStore(store, types.KeyPoolIncentiveRecords(poolId))
+
+	var anys []*codectypes.Any
+	pageRes, err := query.Paginate(incentiveRecordStore, pagination, func(key, value []byte) error {
```
Comment (mattverse) [inline x/concentrated-liquidity/incentives.go:572]: Consider using value returned here instead of key

**Comment by mattverse** [comment]
@stackman27 can you reassign me as reviewer once r4r?

**Comment by stackman27** [comment]
> @stackman27 can this get revived

yea i've been a little stick in this function (https://github.com/osmosis-labs/osmosis/pull/5185/files#diff-8fb13ed7a62b35b89555bdc906798a4353c391186de40eacadac4045c1e4d7bdR599) 


for some reason my `incentivesRecordStore` is not being read and i get this error 
` Wrong incentive prefix, got: [], required [4]` any help is really appreciated

**Pull request merged**